go.mod

```diff
@@ -2,6 +2,11 @@ module github.com/containerd/containerd
 
 go 1.16
 
+require github.com/containerd/containerd/api v0.0.0
+
+// Use the relative local source of the github.com/containerd/containerd/api to build
+replace github.com/containerd/containerd/api => ./api
```

Comment:
Perhaps better to move these together with the other `require` and `replace` rules (otherwise one or the other would be easy to miss).

We should also;

- make sure that github.com/containerd/containerd/api is updated to an _actual version_ (at least when we release), otherwise consumers of containerd will get "random version that we may not have tested with"
- have some CI step that tests both against current `main` version of `api` (which is what this does), and against the version that's specified in `require` (which is what "everyone else" will be using)

Reply:
i've done the first bit (gather the require / replace rules)

+1 to the other 2 points as follow up.